storage/src/main/java/io/delta/storage/S3SingleDriverLogStoreUtil.java

```diff
@@ -0,0 +1,47 @@
+package io.delta.storage;
+
+import com.amazonaws.services.s3.model.ListObjectsV2Request;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.RemoteIterator;
+import org.apache.hadoop.fs.s3a.Listing;
+import org.apache.hadoop.fs.s3a.S3AFileStatus;
+import org.apache.hadoop.fs.s3a.S3AFileSystem;
+import org.apache.hadoop.fs.s3a.S3ListRequest;
+
+import java.io.IOException;
+import java.util.HashSet;
+
+import static org.apache.hadoop.fs.s3a.S3AUtils.ACCEPT_ALL;
+import static org.apache.hadoop.fs.s3a.S3AUtils.iteratorToStatuses;
+
+
+/**
+ * Static utility methods for the S3SingleDriverLogStore.
+ *
+ * Used to trick the class loader so we can use methods of org.apache.hadoop:hadoop-aws without needing to load this as
+ * a dependency for tests in core.
+ */
+class S3SingleDriverLogStoreUtil {
```

Comment:
nit: could you put this into `io.delta.storage.internal`? Could you also rename this to `S3LogStoreUtils` as we are going to use it in `S3DynamoDBLogStore` as well?

Reply:
:100: